src/ape/managers/project.py

```diff
@@ -26,17 +27,31 @@ def __post_init__(self):
             self.path = Path(self.path)
 
         self.dependencies = {
-            manifest.name: manifest
-            for manifest in map(self._extract_manifest, self.config.dependencies)
+            n: self._extract_manifest(n, dep_id) for n, dep_id in self.config.dependencies.items()
         }
 
-    def _extract_manifest(self, manifest_uri: str) -> PackageManifest:
-        manifest_dict = requests.get(manifest_uri).json()
-        # TODO: Handle non-manifest URLs e.g. Ape/Brownie projects, Hardhat/Truffle projects, etc.
-        if "name" not in manifest_dict:
-            raise ProjectError("Dependencies must have a name.")
+    def _extract_manifest(self, dependency_name: str, dependency_id: str) -> PackageManifest:
+        if dependency_id.startswith("https://") or dependency_id.startswith("http://"):
+            manifest_dict = requests.get(dependency_id).json()
+            if "name" not in manifest_dict:
+                raise ProjectError("Dependencies must have a name.")
 
-        return PackageManifest.from_dict(manifest_dict)
+            return PackageManifest.from_dict(manifest_dict)
+        else:
+            # Github dependency
+            try:
+                path, version = dependency_id.split("@")
+            except ValueError:
+                raise ValueError("Invalid Github ID. Must be given as <org>/<repo>@<version>")
+
+            target_path = self._cache_folder / "contracts" / dependency_name
```

Comment:
Don't need "contracts" in cache folder, it is only for build files

Also was wanting to use a cache folder in data folder to store the raw files, and to compile them into a manifest file that would be referenced as a single JSON file